src/qt/treectrl.cpp

```diff
@@ -8,11 +8,438 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+#pragma hdrstop
+#endif
+
 #include "wx/treectrl.h"
+#include "wx/imaglist.h"
+
+#include "wx/qt/private/winevent.h"
+#include "wx/qt/private/treeitemfactory.h"
 
 #include <QtWidgets/QTreeWidget>
+#include <QtWidgets/QHeaderView>
+#include <QtGui/QPainter>
+
+namespace
+{
+struct TreeItemDataQt
+{
+    TreeItemDataQt() : data(NULL)
+    {
+    }
+
+    TreeItemDataQt(wxTreeItemData* data) : data(data)
+    {
+        if ( !registered )
+        {
+            qRegisterMetaTypeStreamOperators<TreeItemDataQt>("TreeItemDataQt");
+            registered = true;
+        }
+    }
+
+    wxTreeItemData *data;
+
+    private:
+    static bool registered;
+};
+
+bool TreeItemDataQt::registered = false;
+Q_DECLARE_METATYPE(TreeItemDataQt)
+
+QDataStream &operator<<(QDataStream &out, const TreeItemDataQt &WXUNUSED(obj))
```

Comment:
Why do we implement these operators in such a useless way? Do we have to implement them even if there is no reasonable way to do it? A comment explaining this would be really welcome here.